src/FormMigration/Controllers/MigrationController.php

```diff
@@ -50,6 +51,17 @@ public function __invoke(DonationForm $formV2)
             ->process($payload)
             ->finally(function(FormMigrationPayload $payload) {
                 $payload->formV3->save();
+
+                $campaignRepository = give(CampaignRepository::class);
+
+                // Associate upgraded form to a campaign
+                if ($campaign = $campaignRepository->getByFormId($payload->formV2->id)) {
+                    $defaultForm = $campaign->defaultForm();
+                    $isDefaultForm = $defaultForm && $defaultForm->id == $payload->formV2->id;
+
+                    $campaignRepository->addCampaignForm($campaign, $payload->formV3->id, $isDefaultForm);
```

Comment:
@alaca We may need to do this in two parts:

1. [Migration] Add the v3 form to the campaign
2. [Transfer] (Maybe) Promote the v3 form to the default form for the campaign

The upgrade to v3 form process happens in two parts, migration then transfer.

Migration creates the v3 form, but isn't (at that point) expected to replace the v2 form.

Transfer is the when the v3 form _replaces_ the v2 form, at which point the donations are re-assigned to the v3 form. This is probably when we should _transfer_ default form status to the v3 form.

Reply:
/cc @glaubersilva @jonwaldstein @JasonTheAdams

Reply:
Good point.

Reply:
I cross-posted to Slack to get more eyes on it, so we can all be on the same page.

Reply:
@alaca I have confirmation from @JasonTheAdams that this needs to be done it two parts.

> [Migration] Add the v3 form to the campaign
> [Transfer] (Maybe) Promote the v3 form to the default form for the campaign

Reply:
@kjohnson resolved https://github.com/impress-org/givewp/pull/7592/commits/70c5d826e5aac917eccd395b22c238629d8cee66